pandas/core/groupby/groupby.py

```diff
@@ -2312,26 +2312,42 @@ def _get_cythonized_result(
             return self._wrap_transformed_output(output)
 
     @Substitution(name="groupby")
-    @Appender(_common_see_also)
     def shift(self, periods=1, freq=None, axis=0, fill_value=None):
         """
         Shift each group by periods observations.
 
+        When freq is not passed, shift the index without realigning the data.
+        If freq is passed, the index will be increased using the periods and the freq.
+
         Parameters
         ----------
         periods : int, default 1
             Number of periods to shift.
         freq : str, optional
             Frequency string.
         axis : axis to shift, default 0
+            Shift direction.
         fill_value : optional
+            The scalar value to use for newly introduced missing values.
+            The default depends on the dtype of *self*.
+            For numeric data, `np.nan` is used.
+            For datetime, timedelta, or period data, etc. **NaT** is used.
+            For extension dtypes, `self.dtype.na_value` is used.
 
             .. versionadded:: 0.24.0
 
         Returns
         -------
         Series or DataFrame
             Object shifted within each group.
+
+        See Also
+        --------
+        Index.shift : Shift values of Index.
+        DatetimeIndex.resample : Shift values of DatetimeIndex.
```

Comment:
Let's remove DatetimeIndex.resample and PeriodIndex.shift as neither are public (this docstring is)

Reply:
sure!